Added coverage flags and test for readNumber

Jonagoo opened the pull request:
Added the coverage flags to measure coverage as well as a test to print it for function readNumber in IterImplForStreaming.java
This should resolve #44

Jonagoo committed:
Added coverage flags and test for readNumber

Added the coverage flags to measure coverage as well as a test to print it for function readNumber in IterImplForStreaming.java
This should resolve #44

Samuelhertzberg commented:
Depending on outcome of #42 this will either be approved or changed

Samuelhertzberg reviewed (changes requested):
make changes according to @Jonagoo comment in #42

Jonagoo committed:
Fixed flags in readNumber

Fixed coverage flags in readNumber to not be flagged as true when falling through from a previous case.
This should resolve #44

apeinot69 committed:
Merge branch 'coverage' into issue44

Resolve conflicts before merging issue44 into coverage
This fix #44
